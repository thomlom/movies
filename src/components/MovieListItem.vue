<template>
  <li class="movie" @click="$router.push(`/${movie.imdbID}`)">
    <poster class="movie__poster" :posterUrl="movie.Poster" :title="movie.Title" />
    <p class="movie__title">{{ movie.Title }}</p>
    <span class="movie__year">{{ movie.Year }}</span>
  </li>
</template>

<script>
import Poster from './Poster'

export default {
  name: 'movie-list-item',
  components: {
    Poster
  },
  props: {
    movie: {
      type: Object,
      required: true
    }
  }
}
</script>

<style lang="scss" scoped>
.movie {
  list-style: none; // remove the bullet

  transition: all 0.5s;

  padding: 2rem;

  &:hover {
    cursor: pointer;
    background-color: var(--blue-darkest);
  }

  &__poster {
    height: 45rem;

    @media screen and (max-width: 600px) {
      height: 40rem;
    }
  }

  &__title {
    margin-top: 1rem;
    margin-bottom: 0.5rem;

    font-size: var(--title-small);
    font-weight: 700;

    color: var(--white);
  }

  &__year {
    font-size: var(--title-small);

    color: var(--grey-light);
  }

  &__poster {
    width: 100%;
    height: 45rem;
    object-fit: cover;
    border-radius: 1.5rem;
    box-shadow: var(--shadow-dark);

    @media screen and (max-width: 600px) {
      height: 40rem;
    }
  }
}
</style>
