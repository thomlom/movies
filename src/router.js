import Vue from "vue";
import Router from "vue-router";

import MovieList from "./components/MovieList";
import MovieDetail from "./components/MovieDetail";

Vue.use(Router);

export const routes = [
  {
    path: "/",
    name: "movie-list",
    component: MovieList
  },
  {
    path: "/:imdbId",
    name: "movie-detail",
    component: MovieDetail,
    props: true
  }
];

export default new Router({
  mode: "history",
  base: process.env.BASE_URL,
  routes
});
