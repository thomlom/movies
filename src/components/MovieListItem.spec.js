import { render } from "@testing-library/vue";
import { routes } from "../router";
import MovieListItem from "./MovieListItem.vue";

test("Redirect to the movie detail page on click", async () => {
  const component = render(MovieListItem, {
    props: {
      movie: {
        Poster: "test.jpg",
        Title: "Avengers",
        Year: 2010
      }
    },
    routes
  });

  expect(component.html()).toMatchSnapshot();
});
