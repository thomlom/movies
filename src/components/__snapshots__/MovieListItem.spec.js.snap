// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redirect to the movie detail page on click 1`] = `
<li class="movie"><img src="test.jpg" class="poster movie__poster" title="Avengers">
  <p class="movie__title">Avengers</p> <span class="movie__year">2010</span>
</li>
`;
